nes-sources/include/Sources/Source.hpp

```diff
@@ -30,20 +32,48 @@ namespace NES
 class Source
 {
 public:
+    class FillTupleBufferResult
+    {
+        explicit FillTupleBufferResult(size_t sizeInBytes) : result(Data{sizeInBytes}) { };
+        FillTupleBufferResult() = default;
+
+        struct EoS
+        {
+        };
+
+        struct Data
+        {
+            size_t sizeInBytes;
+        };
+
+        std::variant<EoS, Data> result = EoS{};
+
+    public:
+        static FillTupleBufferResult eos() { return {}; }
+
+        static FillTupleBufferResult withBytes(size_t sizeInBytes) { return FillTupleBufferResult{sizeInBytes}; }
```

Comment:
```suggestion
        static FillTupleBufferResult EOS() { return {}; }

        static FillTupleBufferResult WithBytes(size_t sizeInBytes) { return FillTupleBufferResult{sizeInBytes}; }
```
I think this looks nicer, but this is not something I am willing to die on the hill for